modules/core/shared/src/main/scala/gem/syntax/TreeMap.scala

```diff
@@ -20,11 +21,52 @@ final class TreeMapCompanionOps(val self: TreeMap.type) extends AnyVal {
     */
   def fromFoldable[F[_], A, B](fab: F[(A, B)])(implicit F: Foldable[F], A: Ordering[A]): TreeMap[A, B] =
     fromList(F.toList(fab))
+
+  /** Combines all the given maps into a single map, where keys common to two or
+    * more maps are the value of the last occurrence in the list.
+    */
+  def join[A: Ordering, B](ms: List[TreeMap[A, B]]): TreeMap[A, B] =
+    ms.foldLeft(TreeMap.empty[A, B]) { case (m0, m1) => m0 ++ m1 }
```

Comment:
We need to be a little careful here because this is a cromulent monoid but it's inconsistent with the default one. I think this is fine for now though.